caffeine/src/test/java/com/github/benmanes/caffeine/cache/TestEdenResize.java

```diff
@@ -0,0 +1,93 @@
+package com.github.benmanes.caffeine.cache;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.Console;
+import java.util.Map;
+
+import static java.lang.Thread.sleep;
+
+public class TestEdenResize {
+    @Test
+    public void TestResizingEden() {
+        Cache<Integer, Integer> cache = Caffeine.newBuilder().maximumSize(1024).build();
+        cache.put(1, 1);
+        Policy.Eviction<Integer, Integer> eviction = cache.policy().eviction().orElse(null);
+        Assert.assertNotNull(cache.getIfPresent(1));
+        Assert.assertTrue("Failed to set percentMain to 0.5", cleanlySetPercentMain(eviction, 0.5d));
+        Assert.assertNotNull(cache.getIfPresent(1));
+        Assert.assertTrue("Failed to set percentMain to 0", cleanlySetPercentMain(eviction, 0d));
+        Assert.assertNotNull(cache.getIfPresent(1));
+        Assert.assertTrue("Failed to set percentMain to 1", cleanlySetPercentMain(eviction, 1d));
+        Assert.assertNotNull(cache.getIfPresent(1));
+    }
+
+    private boolean cleanlySetPercentMain(Policy.Eviction<Integer, Integer> eviction, double percentMain) {
+        Assert.assertNotNull(eviction);
+        try {
+            eviction.setMaximum(eviction.getMaximum(), percentMain);
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    @Test
+    public void TestPercentageMainOutOfBounds() {
+        Cache<Integer, Integer> cache = Caffeine.newBuilder().maximumSize(1024).build();
+        Policy.Eviction<Integer, Integer> eviction = cache.policy().eviction().orElse(null);
+        Assert.assertFalse("Failed to throw for percentMain < 0", cleanlySetPercentMain(eviction, -1d));
+        Assert.assertFalse("Failed to throw for percentMain > 1", cleanlySetPercentMain(eviction, 1.01d));
+    }
+
+    @Test
+    public void Test100PercentWindow() throws InterruptedException {
+        Cache<Integer, Integer> cache = Caffeine.newBuilder().maximumSize(1000).build();
+        Policy.Eviction<Integer, Integer> eviction = cache.policy().eviction().get();
+        eviction.setMaximum(eviction.getMaximum(), 0d);
+        for (int counter = 0; counter < 2000; ++counter) {
+            cache.put(counter, counter);
+        }
+
+        // need to wait for a moment to give the cache time to evict stuff.
+        sleep(1000);
```

Comment:
Wondering if the executor allows us to force this to happen rather than racy sleep.